### PR TITLE
Fix gitbutler/integration nuking itself

### PR DESCRIPTION
**Pull request by Caleb-T-Owens**

### CODE REVIEW

File: crates/gitbutler-operating-modes/src/lib.rs [hunk 1]
```diff
@@ -30,9 +30,9 @@ pub fn in_outside_workspace_mode(ctx: &CommandContext) -> Result<bool> {
...
 
 pub fn assure_outside_workspace_mode(ctx: &CommandContext) -> Result<()> {
-    if in_open_workspace_mode(ctx)? {
+    if in_outside_workspace_mode(ctx)? {
         Ok(())
     } else {
-        bail!("Unexpected state: cannot perform operation on non-integration branch")
+        bail!("Unexpected state: cannot perform operation on integration branch")
     }
 }
```
Comment (Caleb-T-Owens) [inline crates/gitbutler-operating-modes/src/lib.rs:38]: I've got tests written for these functions in a separate branch

File: crates/gitbutler-watcher/src/handler.rs [hunk 2]
```diff
@@ -171,7 +171,10 @@ impl Handler {
...
                     let ctx = CommandContext::open(&project)
                         .context("Failed to create a command context")?;
-                    if in_open_workspace_mode(&ctx)? {
+
+                    // If the user has left gitbutler/integration, we want to delete the reference.
+                    // TODO: why do we want to do this?
+                    if in_outside_workspace_mode(&ctx)? {
                         let mut integration_reference = ctx.repository().find_reference(
                             &Refname::from(LocalRefname::new("gitbutler/integration", None))
                                 .to_string(),
```
Comment (Caleb-T-Owens) [inline crates/gitbutler-watcher/src/handler.rs:180]: I don't have a test for this, but you can bet your socks I'm making one.

**Comment by Caleb-T-Owens** [comment]
@mtsgrd CC

File: crates/gitbutler-operating-modes/src/lib.rs (hunk 1 above)
Comment (Caleb-T-Owens) [reply crates/gitbutler-operating-modes/src/lib.rs:38]: This function is unused and isn't causing any problems, but I wanted to fix it so we don't have a sleeping foot-gun.

I've rewritten it in a seperate PR anyway.

**Comment by mtsgrd** [comment]
Awesome, thanks for fixing!

**Pull request merged**